src/poetry/installation/executor.py

```diff
@@ -68,6 +68,17 @@ def __init__(
         self._use_modern_installation = config.get(
             "installer.modern-installation", True
         )
+        if not self._use_modern_installation:
+            self._io.write_line(
```

Comment:
One thing: Shouldn't we use `write_error_line()` for warnings?

Reply:
some do, some don't: if there is a rule then it is not consistently followed.  The stdout from these commands is not really suitable for parsing anyway so I don't suppose it much matters.

I'm not at the right computer to do anything about this now, feel free do any of tweak it yourself / ask me to do it later / not bother.

Reply:
You are probably right that it is inconsistent and does not matter.

Interestingly enough, I see random test failures on Windows if I write the warning to error out. It's always a PermissionError in https://github.com/python-poetry/poetry/blob/0eedea4e1411667b525c234a363f8b7eb5e8fc4b/tests/conftest.py#L356

but in different tests, for example one or two of

```
tests/console/commands/self/test_add_plugins.py::test_add_with_constraint
tests/console/commands/self/test_add_plugins.py::test_add_with_git_constraint_with_extras
tests/console/commands/self/test_remove_plugins.py::test_remove_installed_package
tests/console/commands/self/test_remove_plugins.py::test_remove_installed_package_dry_run
tests/installation/test_executor.py::test_executor_should_write_pep610_url_references_for_editable_git
```

Maybe, a reason to just keep it as is, even if this is a bit unsettling...